openslides_backend/action/actions/poll/stop.py

```diff
@@ -36,8 +38,98 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
             [
                 "poll_couple_countdown",
                 "poll_countdown_id",
+                "users_enable_vote_weight",
             ],
         )
         if meeting.get("poll_couple_countdown") and meeting.get("poll_countdown_id"):
             self.control_countdown(meeting["poll_countdown_id"], "reset")
+
+        # calculate votescast, votesvalid, votesinvalid
+        voted_ids = poll.get("voted_ids", [])
+        instance["votescast"] = f"{len(voted_ids)}.000000"
+        if not meeting.get("users_enable_vote_weight") or not voted_ids:
+            instance["votesvalid"] = instance["votescast"]
+        else:
+            gmr = GetManyRequest(
+                Collection("user"), voted_ids, [f"vote_weight_${poll['meeting_id']}"]
+            )
+            gm_result = self.datastore.get_many([gmr])
+            users = gm_result.get(Collection("user"), {}).values()
+            instance["votesvalid"] = str(
+                sum(
+                    [
+                        Decimal(
+                            entry.get(f"vote_weight_${poll['meeting_id']}", "1.000000")
+                        )
+                        for entry in users
+                    ]
+                )
+            )
+        instance["votesinvalid"] = "0.000000"
+
+        # set entitled users at stop.
+        instance["entitled_users_at_stop"] = self.get_entitled_user_ids(poll)
         return instance
+
+    def get_entitled_user_ids(self, poll: Dict[str, Any]) -> List[Any]:
```

Comment:
```suggestion
    def get_entitled_user(self, poll: Dict[str, Any]) -> List[<typed dict>]:
```
You return more than ids. Also, please create a typed dict for the return type for better type hints.

Reply:
Whoops, had a typo there, it must be `users` (plural), of course